### DISCUSSION TITLE
How to prevent a template appearing in Visual Studio?

### DISCUSSION THREAD

**Discussion by mrlacey** [dotnet/templating#4772] [Template authoring] [answered]
We have a template that we use as part of a large open-source project.

The intention for this template is only to use it from the command line.

When installed (via a command line) on a machine, Visual Studio lists it as an option in the "New Project" dialog. Is there a way to prevent this template from appearing there?

We want to do this because:
- The template doesn't work when run from within Visual Studio. (aside: I'm not concerned with why it doesn't work--for now.)
- The template is intended to be used to create new projects in a specific folder within the existing solution structure. As the "New Project" wizard allows the selection of the location to create the new project, this could easily end up with the generated project being in the _wrong_ place and so not have the correct (expected) relative paths.

It's going to be much easier (and desirable) for us to prevent the template from appearing in the wizard/dialog than deal with the support requests and consequences of them not working or being created somewhere with the wrong relative paths.

**Comment by phenning** [comment, marked as answer]
You can create an ide.host.json in your .template.config directory to keep the template from showing in the new project dialog. 
```
{
  "unsupportedHosts": [
    { 
      "id": "vs" 
    }
  ]
}
```

**Reply by mrlacey** [reply]
brilliant. thank you. 😄
